Prog/Master School Update Through 2017-18.sas

```diff
@@ -0,0 +1,226 @@
+	ui_id_n = ui_id+0;
+	keep ui_id_n master_school_name dcps status;
+run;
+
+proc sort data = closed_schools; by ui_id_n; run;
+
+
+/*Use Enrollment files for school directory for 2015, 2016 and 2017 */
+proc sort data = enroll.Enroll_15_16 
+	out = schools_15_16_in (rename = (ui_id=ui_id_n)); 
+	by ui_id; 
+run;
+
+proc sort data = enroll.Enroll_16_17 
+	out = schools_16_17_in (rename = (ui_id=ui_id_n)); 
+	by ui_id; 
+run;
+
+proc sort data = enroll.Enroll_17_18 
+	out = schools_17_18_in (rename = (ui_id=ui_id_n)); 
+	by ui_id; 
+run;
+
+/* Merge on location data */
+data schools_15_16;
+	merge schools_15_16_in (in=a) school_locations (in=b) closed_schools (in=c);
+	by ui_id_n;
+	if a or c;
+	
+	year = 2015;
+run;
+
+data schools_16_17;
+	merge schools_16_17_in (in=a) school_locations (in=b) closed_schools (in=c);
+	by ui_id_n;
+	if a or c;
+
+	if ui_id_n = 1042000 then do; x = -76.9901 ; y = 38.9024; end;
+	if ui_id_n = 1094702 then do; x = -77.0155929 ; y = 38.9203028; end;
+	if ui_id_n = 1637436 then do; x = -76.9329012 ; y = 38.9065625; end;
+	if ui_id_n = 2108500 then do; x= -77.0283249 ; y = 38.9405892; end;
+	if ui_id_n = 2108600 then do; x = -77.0414395 ; y = 38.8981258; end;
+	if ui_id_n = 2108700 then do; x = -76.9742016 ; y = 38.8556027; end;
+	if ui_id_n = 2108800 then do; x = -76.9986698 ; y = 38.9278646; end;
+
+	year = 2016;
+run;
+
+data schools_17_18;
+	merge schools_17_18_in (in=a) school_locations (in=b) closed_schools (in=c) ;
+	by ui_id_n;
+	if a or c;
+
+	if ui_id_n = 1042000 then do; x = -76.9901 ; y = 38.9024; end;
+	if ui_id_n = 1094702 then do; x = -77.0155929 ; y = 38.9203028; end;
+	if ui_id_n = 1637436 then do; x = -76.9329012 ; y = 38.9065625; end;
+	if ui_id_n = 2108500 then do; x= -77.0283249 ; y = 38.9405892; end;
+	if ui_id_n = 2108600 then do; x = -77.0414395 ; y = 38.8981258; end;
+	if ui_id_n = 2108700 then do; x = -76.9742016 ; y = 38.8556027; end;
+	if ui_id_n = 2108800 then do; x = -76.9986698 ; y = 38.9278646; end;
+	if ui_id_n = 2108701 then do x = -76.9404947; y = 38.8700043; end;
+	if ui_id_n = 2108900 then do; x = -77.0360009 ; y = 38.9260056; end;
+
+	year = 2017;
+run;
+
+
+/* Stack 2015, 2016 and 2017 dataseets */
+data schools_15_18;
+	set schools_15_16 schools_16_17 schools_17_18;
+
+	if SchoolType = 1 then dcps = 1;
+		else if SchoolType = 2 then dcps = 0;
+
+	aud = total;
+
+	ui_id = put(ui_id_n,z7.);
+	drop ui_id_n;
+
+	if master_school_name = " " then do;
+		master_school_name = School_Name;
+	end;
+
+run;
+
+
+/* Use GInside to geocode X/Y to block */
+proc mapimport out=block_shp_in
+  datafile="L:\Libraries\OCTO\Maps\Census_Blocks__2010.shp";  
+run;
+
+proc sort data=block_shp_in out=block_shp (keep = x y geoid10); 
+	by geoid10;
+run;
+
+goptions reset=global border;
+
+proc ginside includeborder
+  data=schools_15_18
+  map=block_shp
+  out=schools_15_18_join;
+  id geoid10;
+run;
+
+
+/* Use DC Data Macros to convert blocks to standard geos */
+data Schools_00_14_geo;
```

Comment:
This data step does not do anything because *GeoBlk2000* is missing for every record. This is because **Schools_00_14** contains only CLOSED school records.

Reply:
@ptatian thanks. Schools_00_14 should not be keeping only closed schools. Looks like I introduced that error at the end when I was trying to add the closed schools on to the newer years. It's corrected now.